python/lsst/daf/butler/core/schema.py

```diff
@@ -305,9 +305,10 @@ def toSqlAlchemy(self, tableName: str, schema: Schema) -> ForeignKeyConstraint:
         constraint : `sqlalchemy.ForeignKeyConstraint`
             SQLAlchemy version of the foreign key constraint.
         """
+        name = "_".join(["fkey", tableName, self.table] + list(self.target) + list(self.source))
```

Comment:
I think this has a potential to exceed name size limits for some backends.

Reply:
I'm a bit worried about that, too, but @MichelleGower hasn't reported any problems in her tests yet, and I don't think we can guarantee a unique name without including all of these terms, even if they aren't usually necessary.  Given that this isn't really a useful-visible name (except maybe in some pretty low-level debugging contexts), I think the next step is probably trying to hash these terms into something with fixed size, and I'd rather not do that on this ticket.